src/linter/yaml/YamlLinter.ts

```diff
@@ -0,0 +1,83 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import deprecatedLibraries from "../../utils/deprecatedLibs.js";
+import {DataWithPosition, fromYaml, getPosition} from "data-with-position";
+
+interface YamlWithPosInfo extends DataWithPosition {
+	framework?: {
+		libraries?: {
+			name: string;
+		}[];
+	};
+	positionKey?: {
+		end: {
+			column: number;
+			line: number;
+		};
+		start: {
+			column: number;
+			line: number;
+		};
+	};
+}
+
+export default class YamlLinter {
+	#content;
+	#resourcePath;
+	#context: LinterContext;
+
+	constructor(resourcePath: string, content: string, context: LinterContext) {
+		this.#content = content;
+		this.#resourcePath = resourcePath;
+		this.#context = context;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/require-await
+	async lint() {
+		try {
+			// Split Yaml file into part documents by '---' separator
+			const allDocuments: string[] = this.#content.split(/(?:\r?\n|\r|\n)---/g);
```

Comment:
```suggestion
			const allDocuments: string[] = this.#content.split(/(?:\r\n|\r|\n)---/g);
```

Redundant quantifier

Reply:
done